src/nix/hash.cc

```diff
@@ -99,15 +65,78 @@ struct CmdHashBase : Command
     }
 };
 
+/**
+ * `nix hash path`
+ */
+struct CmdHashPath : CmdHashBase
```

Comment:
Now that we have different commands for hashing a file and a fs tree, we should give them a separate documentation rather than the weird pattern-matching that we have right now

Reply:
Which weird pattern matching? Remember that `nix hash file` is deprecated, and `nix hash path` is intended for both files and FS trees.